### ISSUE TITLE
[JIT] Generalize checkTrace() to also compare gradients of parameters, not just inputs

### ISSUE THREAD

**Issue by jamesr66a** [pytorch/pytorch#8453]
Right now `checkTrace()` in test_jit.py only checks gradients for inputs, but does not check gradients for parameters on a traced module. We should generalize this so that we can pass in regular modules and test them

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
